File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        
        int n = grid.size();
        int count = 0;
        
        map<vector<int>, int> mp;
        
        for(auto row: grid)
            mp[row]++;
        
        for(int c=0; c<n; c++){
            vector<int> col(n);
            for(int r=0; r<n; r++){
                col[r] = grid[r][c];
            }
            
            if(mp.find(col) != mp.end())
                count += mp[col];
        }
        
        return count;
    }
};

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.java
class Solution {
    public int equalPairs(int[][] grid) {
        
        int n = grid.length;
        int count = 0;
        
        Map<String, Integer> map = new HashMap<>();
        
        for(int[] row: grid){
            String rowString = Arrays.toString(row);
            map.put(rowString, map.getOrDefault(rowString, 0) + 1);
        }
        
        for(int c=0; c<n; c++){
            int[] col = new int[n];
            for(int r=0; r<n; r++){
                col[r] = grid[r][c];
            }
            
            String colString = Arrays.toString(col);
            count += map.getOrDefault(colString, 0);
        }
        return count;
    }
}